src/dotnet-interactive/CommandLine/CommandLineParser.cs

```diff
@@ -507,15 +507,24 @@ public static void SetUpFormatters(FrontendEnvironment frontendEnvironment, Star
                     {
                         Formatter<ScriptContent>.Register((script, writer) =>
                         {
-                            var fullCode = $@"if (typeof window.createDotnetInteractiveClient === typeof Function) {{
-createDotnetInteractiveClient('{frontedEnvironment.DiscoveredUri.AbsoluteUri}').then(function (interactive) {{
-let notebookScope = getDotnetInteractiveScope('{frontedEnvironment.DiscoveredUri.AbsoluteUri}');
+                            if (!Task.Run(() =>
+                            {
+                                var apiUri = frontedEnvironment.GetApiUriAsync().Result;
+                                var fullCode =
+                                    $@"if (typeof window.createDotnetInteractiveClient === typeof Function) {{
+createDotnetInteractiveClient('{apiUri.AbsoluteUri}').then(function (interactive) {{
+let notebookScope = getDotnetInteractiveScope('{apiUri.AbsoluteUri}');
 {script.ScriptValue}
 }});
 }}";
-                            IHtmlContent content =
-                                PocketViewTags.script[type: "text/javascript"](fullCode.ToHtmlContent());
-                            content.WriteTo(writer, HtmlEncoder.Default);
+                                IHtmlContent content =
+                                    PocketViewTags.script[type: "text/javascript"](fullCode.ToHtmlContent());
+                                content.WriteTo(writer, HtmlEncoder.Default);
+                            }).Wait(apiUriTimeout))
+                            {
+                                throw new TimeoutException("Timeout resolving api uri, please try again.");
```

Comment:
Is this throwing even when the timeout hasn't expired?

Reply:
only if we complete without timeout Task.Wait doesn't throw if we sue a timeout, it returns true if we completed within the timeout interval, false otherwise.